### ISSUE TITLE
Save overwrite confirmation doesn't respond to enter key

### ISSUE THREAD

**Issue by eekee** [OpenTTD/OpenTTD#7347]
## Version of OpenTTD
1.9.0-RC1

## Expected result
Enter and return keys confirm, as is the case for train scrap confirmation.

Personally, I'd be happier with no confirmation at all, but I guess it might be useful for some. Touchscreen users, perhaps.

## Actual result
When you press enter or return, the border of the dialog box flashes.

## Steps to reproduce
Try to save a file with the same name.  I do this often when I'm getting to the point where I want to stop playing; I pause, save, and then I think of the reminders I'll want next time, so I make signs and need to save again.

**Comment by PeterN** [comment]
The focussed edit box of the file list trumps window z-order.

**Comment by eekee** [comment]
This may be the first time that's been a problem.

I suggest making confirmation dialogs always trump edit boxes.

**Comment by eekee** [comment]
... which would annoy people who want the chance to save under a different name.

How about the edit field accepts a second press of return as confirmation?  That would be similar to the text editors I use, but quicker.

**Comment by PeterN** [comment]
I wouldn't expect to be able to type in the file name textbox whilst the confirmation dialog is on screen.

**Comment by eekee** [comment]
Few people would, even including me these days.  In '92, Transport Tycoon seemed to have a highly innovative window system, I was still thinking of it like that, but it's not really so strange as to justify adding weirdness.  

On the other hand, if confirmation always overrides text entry, it would be annoying if, for example, you go to scrap all vehicles in a depot, & when you see the confirmation, you remember you want to make a sign about it.  Not the most terrible case, I suppose.

Why was confirmation added, anyway?  I hope it wasn't just to satisfy feelings of properness.  Did someone actually have an example case of how lack of confirmation hurts them?  I can imagine it would if you regularly build complex things with the game paused, and need to save intermediate stages.  I only recall doing that once in 15 years of playing OpenTTD. :)

**Comment by PeterN** [comment]
If you are selling all vehicles in a depot, then you have already lost the focus of the Edit sign text window when you click on the sell all button. You can just click on that window again, as required already.

**Comment by nielsmh** [comment]
@eekee The confirmation was added to get UI parity with most other software. You're right, it's very seldom TT players will overwrite their saved games. That's more reason: Ask the user to confirm the unusual case.

**Comment by eekee** [comment]
Oh come on, that's the worst reason!  Confirmation was decried all the way back in the 80s by one of the leading software designers of the Apple Lisa and Mac operating systems.  Was he right?  I don't think there's hard data available, but look at the popularity of computers with traditional GUIs.  In the early '00s, if I remember right, 40% of Americans didn't have a computer at home; they didn't want one.  "Market penetration" had plateaued at 60%.  What changed that?  Tablets.  Don't most tablet apps auto-save without confirmation?  Some of course emulate desktop apps, but they seem to be the exception rather than the rule.

Personally, I found traditional GUI horrible to deal with until the worst excesses were worn off -- including a great reduction in confirmation -- and I got somewhat innured to the rest when all the 'modern' 'progressive' GUI programs for Linux started blindly imitating Windows.  (I found it painful and stupid, but that's another story.)  Transport Tycoon wasn't perfect, (menus!) and obviously a little bit limited, but it was well ahead of its time in some ways.  It was the program which showed me that a window system might not be too painful after all.  

In the present, with filesystem snapshots available in perhaps every maintained OS, there's less reason for confirmation than ever.  Windows 10 offers backups at 10 minute intervals!  If I rememer right, it defaults to that.  While restoring from backup might not be the most straightforward thing in the world, at least it doesn't snag your attention when there's nothing wrong.
